### PR TITLE
test: create with correct state and version executions

### PR DESCRIPTION
**Pull request by zoltanpalfi**
Tests for create process instance, sub process, child process.
Closes AAE-1603, AAE-1600

### CODE REVIEW

File: activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    @Before
+    public void setUp() throws Exception {
+        initMocks(this);
+        given(processEngineConfiguration.getClock()).willReturn(clock);
+        given(processEngineConfiguration.getEventDispatcher()).willReturn(eventDispatcher);
+        Context.setProcessEngineConfiguration(processEngineConfiguration);
+    }
+
+    @Test
+    public void should_createProcessInstanceExecution_when_NoInitialVariable() {
```
Comment (ffazzini) [inline activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java]: i don't understand the method name -> "NoInitialVariable" What variable does it refer to?

File: activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        //given
```
Comment (ffazzini) [inline activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java]: given when then comments can be removed

File: activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java (hunk 1 above)
Comment (zoltanpalfi) [reply activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java]: `when_initialVariableNull` is better?

Comment (zoltanpalfi) [reply activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java]: basically I test the case when the on the processInstance creation we are not setting any variable

Comment (ffazzini) [reply activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java]: when_initialVariableNameIsNull ? 
I would like to add "Name" as it is in the method signature
https://github.com/Activiti/Activiti/blob/fc2793f5f896c9d9b0acdb52cf44904b67e882b1/activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImpl.java#L210

Comment (ffazzini) [reply activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java]: is there a special behaviour triggered when initialVariableName is null?

**Pull request merged**